conans/test/unittests/client/build/cpp_std_flags_test.py

```diff
@@ -1,195 +1,261 @@
+
+        self.assertEqual(make_cppstd_flag("clang", "3.5", "11"), '-std=c++11')
+        self.assertEqual(make_cppstd_flag("clang", "3.5", "14"), '-std=c++14')
+        self.assertEqual(make_cppstd_flag("clang", "3.5", "17"), '-std=c++1z')
+
+        self.assertEqual(make_cppstd_flag("clang", "5.0", "11"), '-std=c++11')
+        self.assertEqual(make_cppstd_flag("clang", "5.0", "14"), '-std=c++14')
+        self.assertEqual(make_cppstd_flag("clang", "5.0", "gnu14"), '-std=gnu++14')
+        self.assertEqual(make_cppstd_flag("clang", "5.0", "17"), '-std=c++17')
+
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "5.1", "11")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "5.1", "14")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "5.1", "17")
+
+        self.assertEqual(make_cppstd_flag("clang", "6.0", "11"), '-std=c++11')
+        self.assertEqual(make_cppstd_flag("clang", "6.0", "14"), '-std=c++14')
+        self.assertEqual(make_cppstd_flag("clang", "6.0", "17"), '-std=c++17')
+        self.assertEqual(make_cppstd_flag("clang", "6.0", "20"), '-std=c++2a')
+
+        self.assertEqual(make_cppstd_flag("clang", "7.0", "11"), '-std=c++11')
+        self.assertEqual(make_cppstd_flag("clang", "7.0", "14"), '-std=c++14')
+        self.assertEqual(make_cppstd_flag("clang", "7.0", "17"), '-std=c++17')
+        self.assertEqual(make_cppstd_flag("clang", "7.0", "20"), '-std=c++2a')
+
+        self.assertEqual(make_cppstd_flag("clang", "8", "11"), '-std=c++11')
+        self.assertEqual(make_cppstd_flag("clang", "8", "14"), '-std=c++14')
+        self.assertEqual(make_cppstd_flag("clang", "8", "17"), '-std=c++17')
+        self.assertEqual(make_cppstd_flag("clang", "8", "20"), '-std=c++2a')
 
     def test_clang_cppstd_defaults(self):
-        self.assertEqual(cppstd_default("clang", "2"), "gnu98")
-        self.assertEqual(cppstd_default("clang", "2.1"), "gnu98")
-        self.assertEqual(cppstd_default("clang", "3.0"), "gnu98")
-        self.assertEqual(cppstd_default("clang", "3.1"), "gnu98")
-        self.assertEqual(cppstd_default("clang", "3.4"), "gnu98")
-        self.assertEqual(cppstd_default("clang", "3.5"), "gnu98")
-        self.assertEqual(cppstd_default("clang", "5"), "gnu98")
-        self.assertEqual(cppstd_default("clang", "5.1"), "gnu98")
-        self.assertEqual(cppstd_default("clang", "6"), "gnu14")
-        self.assertEqual(cppstd_default("clang", "7"), "gnu14")
+        with self.assertRaises(ConanException):
+            make_cppstd_default("clang", "2.0")
+        with self.assertRaises(ConanException):
+            make_cppstd_default("clang", "2.1")
+        with self.assertRaises(ConanException):
+            make_cppstd_default("clang", "3.0")
+        with self.assertRaises(ConanException):
+            make_cppstd_default("clang", "3.1")
```

Comment:
```suggestion
```

Failing because of the helper function, not because of the underlying `cppstd_default` being tested here

Reply:
removed